server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ChunkOutputStream.java

```diff
@@ -153,4 +166,994 @@ public void flush() {
         byteSink.emitNext(Tuples.of(bytes, cf), EMIT_FAILURE_HANDLER);
         return cf;
    }
+
+    /**
+     * The ultimate purpose of this wrapper is to complete the {@link CompletableFuture} returned
+     * from {@link #asyncWrite(byte[], int, int)}.  It achieves this with by using the assumption
+     * that reactor-netty will call one of the ByteBuf release methods (e.g. {@link ByteBuf#release()}.
+     */
+    private static class WrappedByteBuf extends ByteBuf {
```

Comment:
This is a pretty long class and, honestly, I had to give me on verifying all calls are right after 30%.  Accordingly, I find it error prone.  But, I do not have any alternative suggestion neither.

Reply:
I have long wanted automatic delegation:)  Anyhow, if we do end up going this route, I will put the key methods at the top (the `release` ones), then all the wrapping calls, and end with all calls that just do delegation.  That will help a little bit.  There are other techniques that make things clearer, but we'd be creating more objects (e.g. a base class that does straight delegation..)

Reply:
How to protect against a scenario where the interface gets a new method with default implementation that returns ByteBuf?

Reply:
If it was an interface, we'd actually be saved; however, we will have this problem since this is actually `extension`.  Honestly, this was just an initial thought on the problem.  I hope there is another solution.

Just in case it's not clear, we can always copy bytes into a new `byte[]` on each call to `asyncWrite`.  That should always work.  Of course, we don't really want to do that..